pkg/server/server_sql.go

```diff
@@ -301,9 +301,13 @@ func newSQLServer(ctx context.Context, cfg sqlServerArgs) (*SQLServer, error) {
 		if cfg.TestingKnobs.JobsTestingKnobs != nil {
 			jobsKnobs = cfg.TestingKnobs.JobsTestingKnobs.(*jobs.TestingKnobs)
 		}
-		td, err := tracedumper.NewTraceDumper(ctx, cfg.InflightTraceDirName, cfg.Settings)
-		if err != nil {
-			log.Errorf(ctx, "failed to initialize trace dumper: %+v", err)
+
+		var td *tracedumper.TraceDumper
+		if cfg.InflightTraceDirName != "" {
+			td, err = tracedumper.NewTraceDumper(ctx, cfg.InflightTraceDirName, cfg.Settings)
```

Comment:
The only error we return from NewTraceDumper is if the passed in dir is "". What do you think of just returning a nil trace dumper from `NewTraceDumper` and simplifying its signature to no longer return an error. I think as it stands the log line below would never get it.

Reply:
yep, you're right, going to do that.